consensus/istanbul/backend/api.go

```diff
@@ -151,36 +179,33 @@ var (
 	errNoBlockNumber           = errors.New("block number is not assigned")
 )
 
-// GetCouncil retrieves the list of authorized validators at the specified block.
-func (api *APIExtension) GetCouncil(number *rpc.BlockNumber) ([]common.Address, error) {
+// GetSnapshot retrieves the state snapshot at a given block. The namespace is klay.
+func (api *APIExtension) GetSnapshot(number *rpc.BlockNumber) (*Snapshot, error) {
 	// Retrieve the requested block number (or current if none requested)
 	var header *types.Header
 	if number == nil || *number == rpc.LatestBlockNumber {
 		header = api.chain.CurrentHeader()
 	} else if *number == rpc.PendingBlockNumber {
-		logger.Trace("Cannot get council of the pending block.", "number", number)
+		logger.Trace("Cannot get snapshot of the pending block.", "number", number)
 		return nil, errPendingNotAllowed
 	} else {
 		header = api.chain.GetHeaderByNumber(uint64(number.Int64()))
 	}
-	// Ensure we have an actually valid block and return the council from its snapshot
+	// Ensure we have an actually valid block and return its snapshot
 	if header == nil {
-		return nil, errNoBlockExist // return nil if block is not found.
+		return nil, errUnknownBlock // return nil if block is not found.
 	}
 
-	// Since v1.6.1, extra.validators represents a list of council
-	// TODO-Klaytn : replace this to the below calculation logic
-	//istanbulExtra, err := types.ExtractIstanbulExtra(header)
-	//if err == nil {
-	//	return istanbulExtra.Validators, nil
-	//} else {
-	//	return nil, errExtractIstanbulExtra
-	//}
+	return api.istanbul.snapshot(api.chain, header.Number.Uint64(), header.Hash(), nil, false)
+}
 
-	// Calculate council list from snapshot
-	snap, err := api.istanbul.snapshot(api.chain, header.Number.Uint64(), header.Hash(), nil, false)
+// GetCouncil retrieves the list of authorized validators at the specified block.
+func (api *APIExtension) GetCouncil(number *rpc.BlockNumber) ([]common.Address, error) {
+	previousBlockNumber := rpc.BlockNumber(number.Int64() - 1)
+	// get the snapshot of the previous block.
+	snap, err := api.GetSnapshot(&previousBlockNumber)
 	if err != nil {
-		logger.Error("Failed to get snapshot.", "hash", header.Hash(), "err", err)
+		logger.Error("Failed to get snapshot.", "hash", snap.Hash, "err", err)
```

Comment:
If the code is changed to call `GetSnapshot`, then what about returning err instead of errInternalError?